### PR TITLE
Align see also item text with target title

### PR DESCRIPTION
**Pull request by Josh-Cena**

### CODE REVIEW

File: files/en-us/web/javascript/guide/regular_expressions/assertions/index.md [hunk 1]
```diff
@@ -293,8 +293,8 @@ console.log(ripeOranges); // [ 'ripe orange A', 'ripe orange C' ]
 
 ## See also
 
-- [Regular expressions](/en-US/docs/Web/JavaScript/Guide/Regular_expressions)
+- [Regular expressions guide](/en-US/docs/Web/JavaScript/Guide/Regular_expressions)
```
Comment (dipikabh) [inline files/en-us/web/javascript/guide/regular_expressions/assertions/index.md:296]: We can keep the description part unlinked and so then the link text will exactly match the title of the target page

```suggestion
- [Regular expressions](/en-US/docs/Web/JavaScript/Guide/Regular_expressions) guide
```

File: files/en-us/web/javascript/reference/classes/index.md [hunk 2]
```diff
@@ -339,5 +339,5 @@ eat(); // global object (in non-strict mode)
 - [Using classes](/en-US/docs/Web/JavaScript/Guide/Using_classes)
 - [`class`](/en-US/docs/Web/JavaScript/Reference/Statements/class)
 - [`class` expression](/en-US/docs/Web/JavaScript/Reference/Operators/class)
-- [Functions](/en-US/docs/Web/JavaScript/Reference/Functions)
+- [Functions reference](/en-US/docs/Web/JavaScript/Reference/Functions)
```
Comment (dipikabh) [inline files/en-us/web/javascript/reference/classes/index.md:342]:
```suggestion
- [Functions](/en-US/docs/Web/JavaScript/Reference/Functions) reference
```
...and also in other occurrences in this PR

File: files/en-us/web/javascript/reference/errors/missing_parenthesis_after_argument_list/index.md [hunk 3]
```diff
@@ -70,4 +70,4 @@ console.log('"Java" + "Script" = "' + "Java" + 'Script"');
 
 ## See also
 
-- [Functions](/en-US/docs/Web/JavaScript/Guide/Functions)
+- [Functions guide](/en-US/docs/Web/JavaScript/Guide/Functions)
```
Comment (dipikabh) [inline files/en-us/web/javascript/reference/errors/missing_parenthesis_after_argument_list/index.md:73]:
```suggestion
- [Functions](/en-US/docs/Web/JavaScript/Guide/Functions) guide
```
 same comment for other occurrences in this PR

**Review by dipikabh — CHANGES_REQUESTED** [review]
Thanks, @Josh-Cena! Just a few nits

**Comment by Josh-Cena** [comment]
@dipikabh FYI, for consistency, I've made all guide articles have the "guide" suffix, and those without are references by default.

**Review by dipikabh — APPROVED** [review]
Thank you!

**Pull request merged**